### DISCUSSION TITLE
[Feature] Docker Secret support

### DISCUSSION THREAD

**Discussion by pixxon** [immich-app/immich#6685] [Feature Request]
### The feature

Hello!

The installation method recommends to use environment variables to configure database connections. However Docker Swarm allows ( and recommends ) to use [Secrets](https://docs.docker.com/engine/swarm/secrets/) to handle sensitive data. This would mean to read the password from the __file__ that is specified by the environment variable.

There are several approaches to handle this, some applications use `file://` prefix to detect that the variable should be used to read from a file. Others use a `_FILE` postfix on the environment variables name, to signal that.

I would recommend instead of handling every variable and determine if they are sensitive or not, to just in general support a read-from-file solution for every environment variable.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
This is already supported, see https://immich.app/docs/install/environment-variables#docker-secrets.

**Reply by pixxon** [reply]
Oh, thank you!

( Sorry, I searched for the usage of the variable in the source code, but had the filter on `.ts` files by accident. )
